guides/source/active_job_basics.md

```diff
@@ -124,8 +124,68 @@ GuestsCleanupJob.perform_later(guest1, guest2, filter: 'some_filter')
 
 That's it!
 
+### Bulk Enqueuing
+
+You can Enqueue multiple jobs at once using [`perform_all_later`][]. Bulk Enqueuing reduces the number of round trips to a queue data store like Redis.
+
+`perform_all_later` is a top-level api on ActiveJob. It accepts instantiated jobs as arguments (note that this is different from `perform_later`). `perform_all_later` does call `perform` under the hood. The arguments passed to `new` will be passed on to `perform` when it's eventually called.
+
+Here is an example calling `perform_all_later` with `GuestCleanupJob` instances:
+
+```ruby
+# Create jobs to pass to `perform_all_later`. The arguments to `new` are
+# passed on to `perform`
+guest_cleanup_jobs = User.pluck(:id).map { |id| GuestsCleanupJob.new(guest_id: id) }
+
+# Will enqueue a seperate job for each instance of `GuestCleanupJob`
+ActiveJob.perform_all_later(guest_cleanup_jobs)
+```
+
+`perform_all_later` logs the number of jobs successfully enqueued, something like `Enqueued 3 jobs to Async (3 GuestsCleanupJob)` if it was called with `ActiveJob.perform_all_later(job1, job2, job3)`. 
+
+The return value of `perform_all_later` is `nil`. Note that this is different from `perform_later`. This may be enhanced to return the number of successfully enqueued job in the future, but for now it's `nil`.
+
+#### Enqueue Multiple ActiveJob Classes
+
+With `perform_all_later` it's also possible to enqueue different ActiveJob class instances in the same call. For example 
+
+```ruby
+class ExportDataJob < ApplicationJob
+  def perform(*args)
+    # Export data
+  end
+end
+
+class NotifyGuestsJob < ApplicationJob
+  def perform(*guests)
+    # Email guests
+  end
+end
+
+# Instantiate job instances
+cleanup_job = GuestsCleanupJob.new(guest)
+export_job = ExportDataJob.new(data)
+notify_job = NotifyGuestsJob.new(guest)
+
+# Enqueues job instances from multiple classes at once 
+ApplicationJob.perform_all_later(cleanup_job, export_job, notify_job)
+```
+#### Callbacks
+
+Note that `perform_all_later` does not trigger callbacks on the individual jobs. This is inline with other ActiveRecord bulk methods. Since callbacks run on individual jobs, they can't take advantage of the bulk nature of this method. Also there isn't a meaningful semantic for something like `around_enqueue` callback.
```

Comment:
(non-blocking, suggestion): The series of short sentences threw me off a bit but maybe this could be personal preference, I'm not sure. Feel free to ignore it if you do not agree. 

```suggestion
`perform_all_later` does not trigger callbacks on the individual jobs (this behavior is in line with other ActiveRecord bulk methods). Callbacks run on individual jobs, hence they can't take advantage of the bulk nature of this method, and there isn't a meaningful semantic for something like `around_enqueue` callback.
```

I'm not sure that I fully understand this portion of the sentence `and there is not a meaningful semantic for something like `around_enqueue` callback.`